central/declarativeconfig/updater/role_updater.go

```diff
@@ -42,19 +42,20 @@ func (u *roleUpdater) Upsert(ctx context.Context, m protocompat.Message) error {
 	return u.roleDS.UpsertRole(ctx, role)
 }
 
-func (u *roleUpdater) DeleteResources(ctx context.Context, resourceIDsToSkip ...string) ([]string, error) {
+func (u *roleUpdater) DeleteResources(ctx context.Context, resourceIDsToSkip ...string) ([]string, int, error) {
```

Comment:
I see a pattern in the implementations of this method. Would it be worth generalizing?

Reply:
Looking closer, there is indeed a pattern, but as many implementations that differ from the pattern than implementations that follow it.

One common pattern emerges for AccessScopes, Roles and PermissionSets.
There is a minor deviation for Groups.
The exceptions for AuthProvider and Notifier are more important.

At the moment, I would prefer to leave the code as it currently is.